backend/src/main/scala/com/softwaremill/adopttapir/config/ConfigReader.scala

```diff
@@ -0,0 +1,45 @@
+package com.softwaremill.adopttapir.config
+
+import cats.data.NonEmptyList
+import cats.data.Reader
+import cats.effect.IO
+import com.softwaremill.adopttapir.config.Config
+import com.softwaremill.adopttapir.http.{Http, HttpApi}
+import com.softwaremill.adopttapir.metrics.VersionApi
+import com.softwaremill.adopttapir.starter.StarterService
+import com.softwaremill.adopttapir.starter.api.StarterApi
+import com.softwaremill.adopttapir.starter.content.ContentService
+import com.softwaremill.adopttapir.starter.files.FilesManager
+import com.softwaremill.adopttapir.starter.formatting.GeneratedFilesFormatter
+import com.softwaremill.adopttapir.template.ProjectGenerator
+import sttp.tapir.server.metrics.prometheus.PrometheusMetrics
+
+type ConfigReader[A] = Reader[Config, A]
+
+object ConfigReader:
```

Comment:
I expected this to read the config file, but in fact this does something else - it performs the wiring of dependencies? I think this might go to `Dependencies` as private members? `Reader` is more of a technical detail, than the main responsibility of the class :)

Reply:
Refactored.